internal/transport/http2_client.go

```diff
@@ -234,6 +239,10 @@ func newHTTP2Client(connectCtx, ctx context.Context, addr TargetInfo, opts Conne
 		maxConcurrentStreams:  defaultMaxStreamsClient,
 		streamQuota:           defaultMaxStreamsClient,
 		streamsQuotaAvailable: make(chan struct{}, 1),
+		deadline:              deadline,
```

Comment:
This is only used inside this one function, and can be removed.

Reply:
Done.